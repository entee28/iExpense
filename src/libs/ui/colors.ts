export default {
  primary5: '#F7FAFF',
  primary10: '#E5EEFF',
  primary20: '#0055FF33',
  primary30: '#99BBFF',
  primary40: 'rgba(0, 85, 255, 0.4)',
  primary50: 'rgba(0, 85, 255, 0.5)',
  primary80: '#3377FF',
  primary100: '#0055FF',
  mono5: '#1A1A1A0D',
  mono4: '#1A1A1A0A',
  mono10: '#1A1A1A1A',
  mono40: '#1A1A1A66',
  mono60: '#1A1A1A99',
  mono70: '#1A1A1AB2',
  mono80: '#1A1A1ACC',
  mono100: '#1A1A1A',
  warning5: '#FFA8000D',
  warning10: '#FFA8001A',
  warning30: '#FFA8004D',
  warning80: '#FFA80080',
  warning100: '#FFA800',
  warning: '#FF6100',
  destructive: '#E60A32',
  destructive80: '#E60A32CC',
  destructive30: '#E60A324D',
  destructive30Hard: '#F8B6C1',
  destructive10: '#E60A321A',
  destructive5: 'rgba(230, 10, 50, 0.05)',
  positive5: '#00B47E0D',
  positive10: '#00B47E1A',
  positive30: '#00B47E4D',
  positive100: '#00B47E',
  white: '#FFFFFF',
  transparent: 'transparent',
  aliceBlue: '#EBF3FF'
}
