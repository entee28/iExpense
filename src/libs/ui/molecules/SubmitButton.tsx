import React from 'react'
import { Box } from '../atoms'
import colors from '../colors'
import { SCREEN_PADDING_HORIZONTAL } from '../constants'
import { Button, ButtonProps } from './Button'

export const SubmitButton = ({ ...props }: ButtonProps) => {
  return (
    <Box
      paddingVertical={16}
      paddingHorizontal={SCREEN_PADDING_HORIZONTAL}
      backgroundColor={colors.white}
      borderTopWidth={1}
      borderTopColor={colors.mono10}>
      <Button {...props} />
    </Box>
  )
}
