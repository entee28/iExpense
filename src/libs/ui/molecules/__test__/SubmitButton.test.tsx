import { render } from '@testing-library/react-native'
import colors from '../../colors'
import { SubmitButton } from '../SubmitButton'

it('Button should render variation styles correctly', () => {
  const primaryBtn = render(
    <SubmitButton
      label="Primary Button"
      variation="primary"
      onPress={() => {}}
      testID="primary_btn"
    />
  )
  const secondaryBtn = render(
    <SubmitButton
      label="Secondary Button"
      variation="secondary"
      onPress={() => {}}
      testID="secondary_btn"
    />
  )
  const tertiaryBtn = render(
    <SubmitButton
      label="Tertiary Button"
      variation="tertiary"
      onPress={() => {}}
      testID="tertiary_btn"
    />
  )
  const disbledBtn = render(
    <SubmitButton
      label="Disabled Button"
      disabled
      onPress={() => {}}
      testID="disabled_btn"
    />
  )

  expect(primaryBtn.getByTestId('primary_btn')).toHaveStyle({
    backgroundColor: colors.primary100
  })
  expect(primaryBtn.getByText('Primary Button')).toHaveStyle({
    color: colors.white
  })

  expect(secondaryBtn.getByTestId('secondary_btn')).toHaveStyle({
    backgroundColor: colors.primary10
  })
  expect(secondaryBtn.getByText('Secondary Button')).toHaveStyle({
    color: colors.primary100
  })

  expect(tertiaryBtn.getByTestId('tertiary_btn')).toHaveStyle({
    backgroundColor: colors.transparent
  })
  expect(tertiaryBtn.getByText('Tertiary Button')).toHaveStyle({
    color: colors.primary100
  })

  expect(disbledBtn.getByTestId('disabled_btn')).toHaveStyle({
    backgroundColor: colors.mono5
  })
  expect(disbledBtn.getByText('Disabled Button')).toHaveStyle({
    color: colors.mono40
  })
})

it('Button should render styles correctly', () => {
  const btn1 = render(
    <SubmitButton
      label="Button 1"
      onPress={() => {}}
      testID="btn1"
      style={{ backgroundColor: 'red' }}
      labelStyle={{ color: 'blue' }}
    />
  )
  const btn2 = render(
    <SubmitButton
      label="Button 2"
      onPress={() => {}}
      testID="btn2"
      variation="primary"
      style={{ backgroundColor: 'red' }}
      labelStyle={{ color: 'blue' }}
    />
  )

  expect(btn1.getByTestId('btn1')).toHaveStyle({
    backgroundColor: 'red'
  })
  expect(btn1.getByText('Button 1')).toHaveStyle({
    color: 'blue'
  })

  expect(btn2.getByTestId('btn2')).toHaveStyle({
    backgroundColor: 'red'
  })
  expect(btn2.getByText('Button 2')).toHaveStyle({
    color: 'blue'
  })
})

it('Button should match snapshot', () => {
  const snapshot = render(
    <SubmitButton
      label="Submit Button"
      onPress={() => {}}
      testID="submit_btn"
      variation="primary"
      style={{ backgroundColor: 'red' }}
      labelStyle={{ color: 'blue' }}
    />
  ).toJSON()

  expect(snapshot).toMatchSnapshot()
})
