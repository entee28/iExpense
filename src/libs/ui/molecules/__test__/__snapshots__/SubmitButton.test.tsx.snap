// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderTopColor": "#1A1A1A1A",
      "borderTopWidth": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 8,
        "opacity": 1,
        "padding": 12,
      }
    }
    testID="submit_btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 14,
            "lineHeight": undefined,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Submit Button
    </Text>
  </View>
</View>
`;
